src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -168,22 +169,24 @@ def download(
         >>> meshing_session.meshing.File.WriteMesh(FileName="write_elbow.msh.h5")
         >>> meshing_session.download(file_name="write_elbow.msh.h5", local_directory="<local_directory_path>")
         """
-        remote_file_name = str(self.fluent_cwd / f"{os.path.basename(file_name)}")
-        local_file_name = None
-        if local_directory:
-            if pathlib.Path(local_directory).is_dir():
-                local_file_name = pathlib.Path(local_directory) / os.path.basename(
-                    file_name
+        files = _get_files(file_name)
+        for file in files:
+            remote_file_name = str(self.fluent_cwd / f"{os.path.basename(file)}")
+            local_file_name = None
+            if local_directory:
+                if pathlib.Path(local_directory).is_dir():
+                    local_file_name = pathlib.Path(local_directory) / os.path.basename(
+                        file
+                    )
+                elif not pathlib.Path(local_directory).is_dir():
+                    local_file_name = pathlib.Path(local_directory)
+            else:
+                local_file_name = pathlib.Path(self.pyfluent_cwd) / os.path.basename(
+                    file
                 )
-            elif not pathlib.Path(local_directory).is_dir():
-                local_file_name = pathlib.Path(local_directory)
-        else:
-            local_file_name = pathlib.Path(self.pyfluent_cwd) / os.path.basename(
-                file_name
-            )
-        if local_file_name.exists() and local_file_name.samefile(remote_file_name):
-            return
-        shutil.copyfile(remote_file_name, str(local_file_name))
+            if local_file_name.exists() and local_file_name.samefile(remote_file_name):
+                return
+            shutil.copyfile(remote_file_name, str(local_file_name))
```

Comment:
This code mixes different approaches for handling file paths, namely pathlib and os.path. While this technically works, it's generally considered better practice to stick to one method to maintain readability and consistency. Since pathlib is more modern, object-oriented, and more readable, it would be good to use it consistently throughout the code. Here's how we can refactor it to use pathlib for path manipulation across the board (caveat: untested code):
```Python
files = _get_files(file_name)
for file in files:
    remote_file_name = self.fluent_cwd / file.name  # pathlib can get the filename directly
    local_file_name = None

    if local_directory:
        local_dir_path = pathlib.Path(local_directory)
        if local_dir_path.is_dir():
            local_file_name = local_dir_path / file.name
        else:
            local_file_name = local_dir_path
    else:
        local_file_name = self.pyfluent_cwd / file.name

    if local_file_name.exists() and local_file_name.samefile(remote_file_name):
        return
    
    shutil.copyfile(remote_file_name, local_file_name)
```

Reply:
Yes, I agree. Thank you.

Reply:
Done. Thank you.